scripts/install-nix-from-closure.sh

```diff
@@ -1,181 +1,700 @@
+{
+checkingRequirements() {
+# This function must have checks only. Without making ANY changes to the system.
 
-# Determine if we should punt to the single-user installer or not
-if [ "$(uname -s)" = "Darwin" ]; then
-    INSTALL_MODE=daemon
-elif [ "$(uname -s)" = "Linux" ] && [ -e /run/systemd/system ]; then
-    INSTALL_MODE=daemon
-else
-    INSTALL_MODE=no-daemon
-fi
+    checkBundle() {
+        if ! [ -e "$self/.reginfo" ]; then
+            error "
 
-# Trivially handle the --daemon / --no-daemon options
-if [ "x${1:-}" = "x--no-daemon" ]; then
-    INSTALL_MODE=no-daemon
-elif [ "x${1:-}" = "x--daemon" ]; then
-    INSTALL_MODE=daemon
-elif [ "x${1:-}" != "x" ]; then
-    (
-        echo "Nix Installer [--daemon|--no-daemon]"
-        echo ""
-        echo " --daemon:    Force the installer to use the Daemon"
-        echo "              based installer, even though it may not"
-        echo "              work."
-        echo ""
-        echo " --no-daemon: Force a no-daemon, single-user"
-        echo "              installation even when the preferred"
-        echo "              method is with the daemon."
-        echo ""
-    ) >&2
-    exit
-fi
+    Installer is incomplete ($self/.reginfo is missing)
+    "
+        fi
+    }
 
-if [ "$INSTALL_MODE" = "daemon" ]; then
-    printf '\e[1;31mSwitching to the Daemon-based Installer\e[0m\n'
-    exec "$self/install-multi-user"
-    exit 0
-fi
+    checkEnv() {
+        if [ "$(id -u)" -eq 0 ]; then
+            # TODO: Reword after scripts integration and option switching becomes clear.
+            notice "
 
-if [ "$(id -u)" -eq 0 ]; then
-    printf '\e[1;31mwarning: installing Nix as root is not supported by this script!\e[0m\n'
-fi
+    Install executed for ROOT.
 
-echo "performing a single-user installation of Nix..." >&2
+    Doing classic Linux package manager mode.
+    In Nix this mode is called: single-user mode for root.
+
+    Nix can do multi-user mode, and manage package trees for users independently.
+    "
+        fi
+
+        # In case USER is not set
+        # Example: running inside container
+        if [ -z "$USER" ]; then
+            notice "
+
+    Variable USER is not set in the environment.
+    "
+            USER="$(id -u -n)"    # id is POSIX
+            print "Detected username: $USER"
+        fi
+
+        if [ -z "$HOME" ]; then
+            error "
+
+    Environment variable HOME is not set.
+    "
+        fi
+
+    }
+
+    checkHome() {
+        if [ ! -e "$HOME" ]; then
+            error "
+
+    Home directory $HOME does not exist.
+    "
+        fi
+
+        if [ ! -d "$HOME" ]; then
+            error "
+
+    Home directory $HOME is not a directory.
+    "
+        fi
+
+        if [ ! -w "$HOME" ]; then
+            error "
+
+    Home directory $HOME is not writable for user $USER. No deal.
+    "
+        fi
+
+        if [ "$(ls -ld "$HOME" | awk '{print $3}')" != "$USER" ]; then # POSIX, no `test -O`
```

Comment:
building release.nix -A binaryTarball.x86_64-linux fails with this warning here:
```
building '/nix/store/9hryb5adyx5qk08c7zcrgy8nza76afpd-nix-binary-tarball-2.1pre6227_a3b7827.drv'...
/nix/store/60bw1jkcv8wfksfhi7sznh1wh66hl5zf-ShellCheck-0.4.7/bin/shellcheck

In /build/install line 282:
        if [ "$(ls -ld "$HOME" | awk '{print $3}')" != "$USER" ]; then # POSIX, no `test -O`
                ^-- SC2012: Use find instead of ls to better handle non-alphanumeric filenames.

builder for '/nix/store/9hryb5adyx5qk08c7zcrgy8nza76afpd-nix-binary-tarball-2.1pre6227_a3b7827.drv' failed with exit code 1
error: build of '/nix/store/9hryb5adyx5qk08c7zcrgy8nza76afpd-nix-binary-tarball-2.1pre6227_a3b7827.drv' failed
```

Reply:
This one in this way is because it is the only POSIX and robust way to get the owner of `$HOME` directory. 
Here paradoxically, `shellcheck` is wrong. We can not use `find` here, because we can not get owner of the directory using `find`.
If someone would introduce ideal way, I would be glad.